src/pages/video.js

```diff
@@ -0,0 +1,62 @@
+import { useParams } from "react-router-dom"
+import Sidebar from "../components/sidebar"
+import VideoCard from "../components/video-card"
+import { useVideos } from "../context/video-context"
+import renderVideoCards from "../functions/renderVideoCards"
+
+const VideoPage = () => {
+    
+    const {videoId} = useParams()
+    
+    const {videoLibrary} = useVideos()
+
+    const {title,creator,description,views,categoryName} = videoLibrary.find((video) => video["_id"] === videoId)
+
+    const videosOfSameCategory = videoLibrary.filter((video) => video.categoryName === categoryName && video["_id"] !== videoId)
```

Comment:
if videosOfSameCategory is being used at other places too then you should consider making it a util function